app/services/monitor_index_job.rb

```diff
@@ -0,0 +1,59 @@
+# Monitor the Done jobs from the SQS queue
+class MonitorIndexJob
+  def initialize
+    @todo_jobs_queue = TodoJobsQueue.new
+    @done_jobs_queue = DoneJobsQueue.new
+    @dead_queue = DeadLetterJobsQueue.new
+    @processed_from_done = 0
+    @aws_asg = OpenStax::Aws::AutoScalingGroup.new(
+      name: Rails.application.secrets[:search_worker_asg_name],
+      region: ENV.fetch('REGION') )
+  end
+
+  def call
+    loop do
+      done_job = @done_jobs_queue.read
+      break if done_job.nil?
+
+      Rails.logger.info("MonitorIndexJob: job #{done_job.class.to_s} #{done_job.to_json} started...")
+
+      done_job.call
+      done_job.cleanup_after_call
```

Comment:
This pattern of calling `call` and then `cleanup_after_call` makes me wonder if the `cleanup_after_call` call should be automatic?  The `BaseIndexJob` has the cleanup_after_call proc in its possession, so we could do...

```ruby
class BaseIndexJob
  def call
    _call
    @cleanup_after_call.try(:call)
  end

  def _call
    raise "Implement _call in the child class!"
  end
end
```

Would this work?

Reply:
i had left it this way in case one wanted to not call cleanup_after_call, keeping the message on the queue for some reason.  but if it's always to be removed, then yea, maybe.

Reply:
@jpslav perhaps this is a candidate for call! and call?  where call! deletes the message after the #call?